services/core/entities/UserEntity.ts

```diff
@@ -7,7 +7,7 @@ export const UserEntity = new Entity({
     PK: { partitionKey: true, hidden: true, prefix: "USERID#" },
     SK: { sortKey: true, hidden: true, default: "ROOT" },
     userId: ["PK", 0, { type: "string", required: true }],
-    apiKey: { type: "string" },
+    apiKey: { type: "binary" },
```

Comment:
interesting type?

Reply:
I first tried storing the apiKey as a type string. But because the encypt command with KMS return a UINT8Array, turning it into a string and storing like that was really weird and I wasn't managing to take the data back and decode it afterwards. I searched a little bit and it seems like it's better to store encrypted data as the binary type in dynamodb: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.DataTypes. (i saw that because of this stackoverflow thread: https://stackoverflow.com/questions/35213615/encrypting-values-using-amazon-kms-storing-retrieving-using-dynamodb-with-lam)
That way it works super well to decrypt the key afterwards. That's why I put the decrypt function in this PR as well, it's not being used yet but it took me some time to get it to work so I wanted to record the result of that investigation for the next person who will have to dev to decryption part